[Bug] Compile failed after syn v2.0 published

nom-derive-impl Cargo.toml
```toml
syn = { version=">=1.0.58", features=["parsing","extra-traits","full"] }
```

cargo tree result:
```shell
├── nom-derive-impl v0.10.0 (proc-macro)
│   │   ├── proc-macro2 v1.0.52 (*)
│   │   ├── quote v1.0.26 (*)
│   │   └── syn v2.0.0
│   │       ├── proc-macro2 v1.0.52 (*)
│   │       ├── quote v1.0.26 (*)
│   │       └── unicode-ident v1.0.8
```

compile error
```shell
error[E0433]: failed to resolve: could not find `NestedMeta` in `syn`
  --> /Users/user/.cargo/registry/src/index.crates.io-6f17d22bba15001f/nom-derive-impl-0.10.0/src/enums.rs:70:42
   |
70 | ...                   syn::NestedMeta::Meta(meta) => match meta {
   |                            ^^^^^^^^^^ could not find `NestedMeta` in `syn`

error[E0277]: the trait bound `MetaAttr: ToTokens` is not satisfied
   --> /Users/user/.cargo/registry/src/index.crates.io-6f17d22bba15001f/nom-derive-impl-0.10.0/src/meta/attr.rs:219:18
    |
219 | impl Spanned for MetaAttr {
    |                  ^^^^^^^^ the trait `ToTokens` is not implemented for `MetaAttr`
    |
    = help: the following other types implement trait `ToTokens`:
              &'a T
              &'a mut T
              Abi
              Abstract
              AndAnd
              AndEq
              AngleBracketedGenericArguments
              Arm
            and 303 others
    = note: required for `MetaAttr` to implement `quote::spanned::Spanned`
    = note: required for `MetaAttr` to implement `syn::spanned::private::Sealed`
note: required by a bound in `syn::spanned::Spanned`
   --> /Users/user/.cargo/registry/src/index.crates.io-6f17d22bba15001f/syn-2.0.0/src/spanned.rs:96:20
    |
96  | pub trait Spanned: private::Sealed {
    |                    ^^^^^^^^^^^^^^^ required by this bound in `Spanned`

error[E0433]: failed to resolve: use of undeclared type `LifetimeDef`
   --> /Users/user/.cargo/registry/src/index.crates.io-6f17d22bba15001f/nom-derive-impl-0.10.0/src/gen/generator.rs:145:42
    |
145 |             .push(GenericParam::Lifetime(LifetimeDef::new(lft.clone())));
    |                                          ^^^^^^^^^^^
    |                                          |
    |                                          use of undeclared type `LifetimeDef`
    |                                          help: a struct with a similar name exists: `Lifetime`
```

--- a/nom-derive-impl/Cargo.toml
+++ b/nom-derive-impl/Cargo.toml
@@ -27,7 +27,7 @@ include = [
 proc-macro = true
 
 [dependencies]
-syn = { version=">=1.0.58", features=["parsing","extra-traits","full"] }
+syn = { version="1.0.58", features=["parsing","extra-traits","full"] }
 quote = "1.0"
 proc-macro2 = "1.0"
 

